### PR TITLE
lib: make GovernorError pub

### PR DESCRIPTION
**Pull request by ernestas-poskus**
Making GovernorError pub. I need to match inner error types and have a custom error handling. 
In my case I use https://www.rfc-editor.org/rfc/rfc7807 https://crates.io/crates/http-api-problem to return errors in this format.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -44,12 +44,10 @@ where
 }
 
 /// https://stegosaurusdormant.com/understanding-derive-clone/
-impl<K: KeyExtractor, M: RateLimitingMiddleware<QuantaInstant>> Clone
-    for GovernorLayer<'_, K, M>
-{
+impl<K: KeyExtractor, M: RateLimitingMiddleware<QuantaInstant>> Clone for GovernorLayer<'_, K, M> {
```
Comment (ernestas-poskus) [inline src/lib.rs:47]: unrelated change *rustfmt*

**Comment by benwis** [comment]
Thanks for the PR! Interesting, I have no problems with that being public. Happy Holidays!
